app/feature/feature-changeaddress/src/main/kotlin/com/hedvig/android/feature/changeaddress/destination/selecthousingtype/SelectHousingTypeViewModel.kt

```diff
@@ -0,0 +1,136 @@
+package com.hedvig.android.feature.changeaddress.destination.selecthousingtype
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import com.hedvig.android.core.ui.ValidatedInput
+import com.hedvig.android.feature.changeaddress.data.ChangeAddressRepository
+import com.hedvig.android.feature.changeaddress.data.HousingType
+import com.hedvig.android.feature.changeaddress.destination.selecthousingtype.SelectHousingTypeEvent.ClearNavigationParameters
+import com.hedvig.android.feature.changeaddress.destination.selecthousingtype.SelectHousingTypeEvent.DismissErrorDialog
+import com.hedvig.android.feature.changeaddress.destination.selecthousingtype.SelectHousingTypeEvent.DismissHousingTypeErrorDialog
+import com.hedvig.android.feature.changeaddress.destination.selecthousingtype.SelectHousingTypeEvent.SelectHousingType
+import com.hedvig.android.feature.changeaddress.destination.selecthousingtype.SelectHousingTypeEvent.SubmitHousingType
+import com.hedvig.android.feature.changeaddress.navigation.SelectHousingTypeParameters
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import hedvig.resources.R
+
+internal class SelectHousingTypeViewModel(changeAddressRepository: ChangeAddressRepository) :
+  MoleculeViewModel<SelectHousingTypeEvent, SelectHousingTypeUiState>(
+    initialState = SelectHousingTypeUiState(),
+    presenter = SelectHousingTypePresenter(changeAddressRepository),
+  )
+
+internal class SelectHousingTypePresenter(private val changeAddressRepository: ChangeAddressRepository) :
+  MoleculePresenter<SelectHousingTypeEvent, SelectHousingTypeUiState> {
+  @Composable
+  override fun MoleculePresenterScope<SelectHousingTypeEvent>.present(
+    lastState: SelectHousingTypeUiState,
+  ): SelectHousingTypeUiState {
+    var dataLoadIteration by remember {
+      mutableIntStateOf(0)
+    }
+
+    var currentState by remember {
+      mutableStateOf(lastState)
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        DismissErrorDialog -> currentState = currentState.copy(errorMessage = null)
+
+        DismissHousingTypeErrorDialog -> currentState = currentState.copy(errorMessageRes = null)
+
+        is SelectHousingType -> {
+          currentState = currentState.copy(housingType = ValidatedInput(input = event.housingType))
+        }
+
+        SubmitHousingType -> {
+          if (!currentState.housingType.isPresent) {
+            currentState = currentState.copy(
+              errorMessageRes = R.string.CHANGE_ADDRESS_HOUSING_TYPE_ERROR,
+            )
+          } else {
+            dataLoadIteration++
```

Comment:
Same concept here, since this is in fact submitting something, I would prefer to have a name that describes that